packages/init/src/utils/tracker.ts

```diff
@@ -89,23 +95,29 @@ export class InitTracker {
 
   identifyUser(userId: string, intercomHash: string): void {
     this.#userId = userId;
-    this._identifyEvent(userId, intercomHash);
+    return this._identifyEvent(userId, intercomHash);
   }
 
-  trackDownloadLibrary(library: string): void {
-    this._trackEvent(EventType.DownloadLibrary, { library });
+  trackDownloadLibrary(library: string): Promise<void> {
+    return this._trackEvent(EventType.DownloadLibrary, { library });
   }
 
-  trackInitIdentify(repoDomain: string | undefined): void {
-    this._trackEvent(EventType.InitIdentify, { repo: repoDomain });
+  trackInitIdentify(repoDomain: string | undefined): Promise<void> {
+    if (repoDomain) this.#repository = repoDomain;
```

Comment:
modifying the Repo should be a function on it's own, we don't want to have to do that for every event.

Reply:
but it can be passed through these events, if it is it's own function. It'll have to be called a few times